service/storage_service.cc

```diff
@@ -6835,8 +6966,12 @@ future<join_node_response_result> storage_service::join_node_response_handler(jo
 void storage_service::init_messaging_service(bool raft_topology_change_enabled) {
     _messaging.local().register_node_ops_cmd([this] (const rpc::client_info& cinfo, node_ops_cmd_request req) {
         auto coordinator = cinfo.retrieve_auxiliary<gms::inet_address>("baddr");
-        return container().invoke_on(0, [coordinator, req = std::move(req)] (auto& ss) mutable {
-            return ss.node_ops_cmd_handler(coordinator, std::move(req));
+        auto coordinator_host_id = cinfo.retrieve_auxiliary_opt<locator::host_id>("host_id");
+        if (!coordinator_host_id) {
+            throw std::runtime_error("Coordinator host_id not found");
```

Comment:
Which Scylla release was the RPC sender's host_id introduced in?

Reply:
It's `5.4` https://github.com/scylladb/scylladb/commit/95c726a8df5fd132a75e35bc2393024dda8c83d8

Reply:
You can't make something mandatory if it was introduced in the previous version, our threshold is two years.

Reply:
@avikivity Do you mean we can skip over 2 years of releases during upgrades? That sounds crazy. Definitely not something we support or test. There are no upgrade guides in our docs that skip a single version even, at least of OSS. In Enterprise there is some version skipping though (we e.g. upgrade from 2022.x to 2023.y)

We may also merge this PR after branching 5.5, so it would end up only in 6.0.

Reply:
Also, suppose this code ends up in 5.5

Then it would only cause problems if someone was doing rolling upgrade from 5.2 to 5.5, and before they finished, they would trigger a topology change on a 5.2 node.

I don't think we should worry about it:
- first of all, they should be upgrading from 5.4, not from 5.2
- second, they shouldn't be doing topology changes
- if they really must do a topology change, they can do it from a 5.4 node, which will send the host_id

Reply:
If it only limits topology changes during upgrade, maybe we can consider it, though it's not very good. We can have a node failure during upgrade.